### ISSUE TITLE
Should the dataverse.payara.timeout value be used to construct the systemd unit TimeoutStartSec value?

### ISSUE THREAD

**Issue by atc0005** [GlobalDataverseCommunityConsortium/dataverse-ansible#35]
I see where the value is used here:

https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible/blob/4010e3b0324ca0ef01db20cbad06a0d17372bcd7/tasks/dataverse-postinstall.yml#L22-L26

but I didn't find it used here:

https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible/blob/45402f9590815b974a82dd247dc288b974c0050e/templates/payara.service.j2#L5-L12

When installing Dataverse within a CentOS 7 LXD container earlier today the playbook finished without complaint, but the daemon failed to startup as expected. It wasn't until I modified the `/usr/lib/systemd/system/payara.service` file like so that it would startup properly:

```diff
-TimeoutStartSec=180s
+TimeoutStartSec=480s
```

What do you think about modifying the `templates/payara.service.j2` file like so?

```diff
-TimeoutStartSec=180s
+TimeoutStartSec={{ dataverse.payara.timeout }}s
```

**Comment by donsizemore** [comment]
now that i'm back on an actual computer (and have had my coffee) i see there are two timeouts, as you asked:

```
    timeout: 180
    request_timeout: 1800
```

and indeed, payara.timeout should be used to construct the systemd unit file from template. will fix this next.

I still see no reason Dataverse should take more than 3 minutes to launch, though... is it dumping much into server.log?

**Comment by atc0005** [comment]
> @donsizemore: I still see no reason Dataverse should take more than 3 minutes to launch, though... is it dumping much into server.log?

I hope to work on this more today (there are some other items already in motion), but if I'm able, I'll take a look at the log file you referenced and report back.

> @donsizemore: indeed, payara.timeout should be used to construct the systemd unit file from template. will fix this next.

Great, thanks!

**Comment by atc0005** [comment]
It took an hour to complete (I didn't time it before), but using the latest from the `master` branch without any local changes to `default/main.yml` the process completed without issue and `payara` launches without issue when bringing the LXD container online. This is with the 180s timeout value.

**Comment by donsizemore** [comment]
@atc0005 default/main.yml on the master branch doesn't compile the warfile or run integration tests. Really the only time-consuming step is R module compilation and possibly software component download depending on your network connection. An hour strikes me as more than three times the length of a similar run on my test hardware - an 8 year-old hand-me-down PC running Ubuntu.

I corrected the systemd template: https://github.com/GlobalDataverseCommunityConsortium/dataverse-ansible/commit/0225a507ff6e759421ae4cc4cca8f2b9131d75ef

**Comment by atc0005** [comment]
> Really the only time-consuming step is R module compilation

I do remember that being a very intensive process on my box, so it is likely that is where the bulk of the time went.

> An hour strikes me as more than three times the length of a similar run on my test hardware - an 8 year-old hand-me-down PC running Ubuntu.

Thanks for noting that. I didn't have much of a comparison for the time required before. 

> I corrected the systemd template: 0225a507ff6e759421ae4cc4cca8f2b9131d75ef

Thanks!
